src/io/router_example.rs

```diff
@@ -123,14 +123,15 @@ mod test {
             },
             Some(SocketAddrV6::new(Ipv6Addr::LOCALHOST, 8080, 0, 0)),
         );
-        let router2: RouterBuilder<ExampleRouterHandler> = RouterBuilder::new(
+        let mut router2: RouterBuilder<ExampleRouterHandler> = RouterBuilder::new(
             ExampleRouterHandler {
                 debug_out: debug_out2.clone(),
             },
             Some(SocketAddrV6::new(Ipv6Addr::LOCALHOST, 8081, 0, 0)),
         );
 
         tokio::spawn(async move {
+            router2.bind().await?;
             router2.listen().await?;
             Ok(())
```

Comment:
bind now returns the ip/port and later call to listen() must be made to actually listen. This way implementers of router (such as read shard) are able to know the port they are listening on while still supporting dynamic port assignment.